src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+---
+title: Core API
+layout: default
+category: Infusion
+---
+
+# Infusion Core API
+
+This file documents some low-level APIs which are provided by Infusion at the JavaScript level. Note that since
+Infusion's more overarching aim is to enable declarative programming based on JSON structures, by means
+of its [IoC](FrameworkConcepts.md#ioc) system, the documentation in this file doesn't cover very much
+of Infusion's real function. Please consult pages on [Component Grades](#ComponentGrades.md), 
+[Component Configuration Options](#ComponentConfigurationOptions.md), the [Infusion Event System](#InfusionEventSystem.md)
+, etc. for coverage of these wider topics.
+
+## Type checking
+
+### fluid.isValue(value)
+
+* `value {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied value is other than `null` or `undefined`.
+
+### fluid.isPrimitive(value)
+
+* `value {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied value is of a primitive JavaScript type (recognized: `String`, `Boolean`, `Number`, `Function`, and falsy values)
+
+### fluid.isArrayable(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied value can be treated as an array. The strategy used is an optimised
+approach taken from an earlier version of jQuery - detecting whether the `toString()` version
+of the object agrees with the textual form `[object Array]`, or else whether the object is a
+jQuery object (the most common source of "fake arrays").
+
+### fluid.isPlainObject(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied object is a plain JSON-forming container - that is, it is either a plain Object
+or a plain Array. 
+
+### fluid.typeCode(totest)
+
+* `totest {Any}`
+* Returns: `{String}`
+
+Returns `primitive`, `array` or `object` depending on whether the supplied object has
+one of those types, by use of the `fluid.isPrimitive`, `fluid.isPlainObject` and `fluid.isArrayable` utilities
+
+### fluid.isDOMNode(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied object is a DOM node.
+
+### fluid.isDOMish(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied object is a DOM node or a jQuery object.
+
+### fluid.isUncopyable(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if it would be unwise to copy the supplied object. The framework will not attempt to copy any material which
+fails this test. An object which passes `fluid.isPrimitive`, `fluid.isDOMish` or fails `fluid.isPlainObject` will pass this test.
+
+### fluid.isComponent(totest)
+
+* `totest {Any}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied object is an Infusion component.
+
+### fluid.isDestroyed(component)
+
+* `component {Component}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied Infusion component has been destroyed.
```

Comment:
The word 'destroyed' here might benefit from being a link to the component lifecycle page